framework/keyring-interface.md

```diff
@@ -36,37 +36,48 @@ in this document are to be interpreted as described in [RFC 2119](https://tools.
 
 ### OnEncrypt
 
-This interface takes [encryption materials](#structures.md#encryption-materials) as input
-and MAY modify it with any of the following behaviors:
+The following inputs are REQUIRED:
+
+- [Algorithm Suite](#algorithm-suites.md)
+- [Encryption Context](#encryption-context)
+
+The following inputs are OPTIONAL:
+
+- [Plaintext Data Key](#structures.md#plaintext-data-key)
+
+This interface MAY return [data key materials](#structures.md#data-key-materials) appropriate for the request. If it does return this output
+and a plaintext data key was provided as input, the output must specify the same plaintext data key.
+
+The output of this interface is driven by two behaviours:
 
 - [Generate data key](#generate-data-key)
 - [Encrypt data key](#encrypt-data-key)
 
 If this keyring attempted any of the above behaviors, and successfully completed those behaviors,
-it MUST output the modified [encryption materials](#structures.md#encryption-materials).
+it MUST output the result as [data key materials](#structures.md#data-key-materials).
 
-If the keyring did not attempt any of the above behaviors, it MUST output the
-[encryption materials](#structures.md#encryption-materials) unmodified.
+If the keyring did not attempt any of the above behaviors, it MUST produce no output.
 
 #### Generate Data Key
 
-If the [encryption materials](#structures.md#encryption-materials) do not contain a plaintext data key,
+If the input does not include a plaintext data key,
 OnEncrypt MAY generate a data key.
-If the encryption materials contain a plaintext data key, OnEncrypt MUST NOT generate a data key.
+If the input does include a plaintext data key, OnEncrypt MUST NOT generate a data key.
 
-Generate Data Key MAY modify the following fields in the [encryption materials](#structures.md#encryption-materials):
+Generate Data Key MAY populate the following fields in the [data key materials](#structures.md#data-key-materials):
 
 - [plaintext data key](#structures.md#plaintext-data-key)
 - [keyring trace](#structures.md#keyring-trace)
 
 To perform this behavior, the keyring generates a [plaintext data key](#structures.md#plaintext-data-key)
-and sets the resulting plaintext data key on the [encryption materials](#structures.md#encryption-materials).
+and includes the resulting plaintext data key in the [data key materials](#structures.md#data-key-materials)' 
+[data key materials](#structures.md#data-key-materials).
```

Comment:
"includes the data key in the data key materials' data key materials"???

I have no idea what that is supposed to mean...do the data key materials separately contain data key materials?? At best that is confusing.

Reply:
Stupid typo, mea cupla. Fixed.